src/spectral/utils/spectral-validator.js

```diff
@@ -0,0 +1,74 @@
+const MessageCarrier = require('../../plugins/utils/messageCarrier');
+const config = require('../../cli-validator/utils/processConfiguration');
+const { isOpenApiv2, isOpenApiv3 } = require('@stoplight/spectral');
+// default spectral ruleset file
+const defaultSpectralRuleset =
+  __dirname + '/../rulesets/.defaultsForSpectral.yaml';
+
+const parseResults = function(results, debug) {
+  const messages = new MessageCarrier();
+
+  if (results) {
+    for (const validationResult of results) {
+      if (validationResult) {
+        const code = validationResult['code'];
+        const severity = validationResult['severity'];
+        const message = validationResult['message'];
+        const path = validationResult['path'];
+
+        if (typeof severity === 'number' && code && message && path) {
+          if (code === 'parser') {
+            // Spectral doesn't allow disabling parser rules, so don't include them
+            // in the output (for now)
+            continue;
+          }
+          // Our validator only supports warning/error level, so only include
+          // those validation results (for now)
+          if (severity === 1) {
+            //warning
+            messages.addMessage(path, message, 'warning');
+          } else if (severity === 0) {
+            //error
```

Comment:
Very picky style comment, comments should have a space after the "//" here and on line 28. It doesn't really matter but I'm surprised eslint didn't complain